benchmarking/caching/simulation_cache.py

```diff
@@ -0,0 +1,115 @@
+import json
+import os
+from datetime import datetime
+from os import path
+from typing import List, Dict, Any
+import git
+from api.models.team_set import TeamSet
+
+
+class SimulationCache:
+    """
+    This class is used to cache the results of a simulation, which is a List[TeamSet], where each element of the list is a different run.
+    All TeamSets should have been generated by the same algorithm (with the same settings).
+    It uses the cache_key to generate a file with the results of the simulation.
+    """
+
+    def __init__(self, cache_key: str) -> None:
+        self.cache_key = cache_key
+        self._data = None
```

Comment:
I'm not a fan not being able to type this, but if I do, it generates a warning about assigning None to type Dict[str, Any]